<launch>
    <!--   task manager, the clients for move base and line track -->
    <node pkg="cashmerebot" type="task_manager.py" name="task_manager" output="screen"/>

    <!--   perception -->
    <include file="$(find cashmerebot)/launch/perception.launch">
        <arg name="run_perception_only" value="false"/>
    </include>

    <!--   manipulation server -->
    <node pkg="cashmerebot" type="manipulation_server.py" name="manipulation_server" output="screen" />

    <!--   conveyor motion server -->
    <node pkg="cashmerebot" type="conv_cmd_server.py" name="conv_cmd_server" output="screen" />

    <!--   ur5e moveit -->
    <include file="$(find cashmerebot)/launch/ur5e_moveit.launch">
        <arg name="use_ur5e_moveit_rviz" value="false"/>
    </include>

<!--     moveit mesh_filter DepthSelfFiltering is not properly handling the urdf file for all links poses. when showing  -->
<!--     filtered/model, links poses of collision links in urdf are all with wrong orientation !!! -->
<!--     <node pkg="nodelet" type="nodelet" name="depth_self_filter_nodelet" args="manager" output="screen"/> -->
<!--     <node pkg="nodelet" type="nodelet" name="mesh_filter" args="load mesh_filter/DepthSelfFiltering depth_self_filter_nodelet" output="screen"> -->
<!--         <remap to="/front_depth_cam/image_raw" from="/depth"/> -->
<!--         <remap to="/front_depth_cam/camera_info" from="/camera_info"/> -->
<!--         <param name="padding_offset" value="0.005" /> -->
<!--     </node> -->

    <!--   rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find cashmerebot)/rviz/task.rviz"/>
</launch>
